device.yml

```diff
@@ -0,0 +1,116 @@
+%YAML 1.1
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/harp-tech/reflex-generator/main/schema/device.json
+device: InputExpander
+whoAmI: 1106
+firmwareVersion: "2.2"
+hardwareTargets: "1.2"
+registers:
+  AuxInPort:
+    address: 32
+    type: U8
+    access: Event
+    maskType: AuxiliaryInputs
+    description: Reports the state of the auxiliary inputs.
+  AuxInRisingEdge:
+    address: 33
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables rising edge detection on the auxiliary inputs.
+  AuxInFallingEdge:
+    address: 34
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables falling edge detection on the auxiliary input port.
+  DigitalInput:
+    address: 35
+    type: U16
+    access: Event
+    length: 2
+    payloadSpec:
+      DigitalInputState:
+        offset: 0
+        maskType: DigitalInputs
+        description: Reports the state of all digital inputs in the port.
+      DigitalInputChanged:
+        offset: 1
+        maskType: DigitalInputs
+        description: Reports which digital inputs changed state in the port.
+    description: Reports the state of the digital inputs.
+  DigitalInputEnableRisingEdge:
+    address: 36
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables rising edge detection on the digital input port.
+  DigitalInputFallingEdge:
+    address: 37
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables falling edge detection on the digital input port.
+  InputSamplingMode:
+    address: 38
+    type: U8
+    access: Write
+    maskType: InputSamplingConfig
+    description: Configures the input sampling mode.
+  EncoderSamplingMode:
+    address: 39
+    type: U8
+    access: Write
+    maskType: EncoderSamplingConfig
+    description: Configures the rotary encoder sampling mode.
+  EncoderData:
+    address: 40
+    type: S16
+    access: Event
+    description: Reports the value of the current read from the rotary encoder.
```

Comment:
"current" or "latest"?

Also checking parity with output expander we used a register called `MagneticEncoder` there: https://github.com/harp-tech/device.outputexpander/blob/3441db0445992929b0db304b7f14592ecc378922/device.yml#L279

Both seem fine to me but just trying to spot possible naming conventions so that interfaces might be made more consistent and predictable in the future.

Reply:
I think when i did this schema i was following the harp behavior
https://github.com/harp-tech/device.behavior/blob/3ff429b3e19bee1c451059667a372b393c7d10bd/device.yml#L66

Given the choice, i think this one should have parity with the sister output expander board.